sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -544,6 +597,42 @@ private[sql] class HDFSBackedStateStoreProvider extends StateStoreProvider with
     result
   }
 
+  private def loadMap(startVersion: Long, endVersion: Long): HDFSBackedStateStoreMap = {
+
+    val (result, elapsedMs) = Utils.timeTakenMs {
+      val startVersionMap = Option(loadedMaps.get(startVersion)) match {
+        case Some(value) =>
+          loadedMapCacheHitCount.increment()
```

Comment:
I feel like we should provide some context as to why we are updating the cache here or accounting towards the cache. if its going to evict legit entries, I wonder whether we should just treat this load as an isolated/independent load operation ?

Reply:
Since the existing state data source also updates these metrics, I am reusing the logic. However, I also think that the two use cases (streaming query and data source) should be isolated. Maybe this can be address by future PR? This also applies to the log warning saying the version is not yet loaded ...

Reply:
State data source initializes the provider instead of sharing the same with streaming query. It was an initial implementation but I roughly remember we fixed that.